zeus/tests/test_poll_views.py

```diff
@@ -0,0 +1,164 @@
+        assert r.status_code == 200
+
+
+class TestPollViews(TestElectionBase):
+
+    def setUp(self):
+        super(TestElectionBase, self).setUp()
+        self.c.post(self.locations['login'], self.login_data)
+        self.election = self.get_election(name="election", voting_starts_at=datetime.date.today(),
+                                     voting_ends_at=datetime.date.today() + datetime.timedelta(days=1),
+                                     trial=True)
+        user = User.objects.get()
+        self.election.admins.add(user)
+        self.election.save()
+
+        self.local_verbose = os.environ.get('ZEUS_TESTS_VERBOSE', None)
+
+    def test_poll_questions(self):
+        poll = self.create_poll(election=self.election, name="poll")
+
+        response = self.c.get('/elections/{}/polls/{}/questions'.format(self.election.uuid, poll.uuid))
+        assert response.status_code == 302
+        self.assertRedirects(response, '/elections/{}/polls/{}/questions/manage'.format(self.election.uuid, poll.uuid),
+                             fetch_redirect_response=True)
+
+        self.e_uuid = self.election.uuid
+        self.submit_voters_file()
+        voter = Voter.objects.all()[0]
+        url = voter.get_quick_login_url()
+
+        self.c.post(self.locations['logout'])
+
+        self.voter_login(voter, url)
+        response = self.c.get('/elections/{}/polls/{}/questions'.format(self.election.uuid, poll.uuid))
+
+        assert response.status_code == 200
+        module = poll.get_module()
+        tpl = getattr(module, 'questions_list_template', 'election_poll_questions')
+
+        self.assertTemplateUsed(response, tpl + '.html')
+
+    def test_poll_add_batch_file(self):
+        self.election.linked_polls = True
+        self.election.save()
+        self.e_uuid = self.election.uuid
+        self.create_poll(self.election, name="linked_poll")
+        poll = Poll.objects.all()[0]
+
+        with open(os.path.join(os.path.dirname(__file__),
+                               'test_sample_survey_for_linked_election.yml')) as batch_file:
+            response = self.c.post('/elections/{}/polls/add'.format(self.election.uuid),
+                                   {
+                                       'batch_file': batch_file,
+                                   })
+
+        assert response.status_code == 302
+        self.assertRedirects(response, '/elections/{}/polls/'.format(self.election.uuid))
+
+        # todo: invalid contents
+
+    def test_poll_remove(self):
+        self.election.linked_polls = True
+        self.election.save()
+        self.e_uuid = self.election.uuid
+        self.create_poll(self.election, name="poll")
+        poll = Poll.objects.all()[0]
+
+        response = self.c.post('/elections/{}/polls/{}/remove'.format(self.election.uuid, poll.uuid))
+
+        assert response.status_code == 302
+        self.assertRedirects(response, '/elections/{}/polls/'.format(self.election.uuid))
```

Comment:
This is a good place for an assertion that a pool got removed `assert not Pool.objects.filter(name="poll").exists()`.